api-metastore/src/main/java/org/zalando/nakadi/service/AvroSchemaCompatibility.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.avro.SchemaCompatibility.SchemaPairCompatibility;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.domain.CompatibilityMode;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class AvroSchemaCompatibility {
+
+    private static BiFunction<Schema, Schema, SchemaPairCompatibility> backwardCompatibleFn = (reader, writer) ->
+            SchemaCompatibility.checkReaderWriterCompatibility(reader, writer);
+
+    private static BiFunction<Schema, Schema, SchemaPairCompatibility> forwardCompatibleFn = (reader, writer) ->
+            SchemaCompatibility.checkReaderWriterCompatibility(writer, reader);
+
+    private static BiFunction<Schema, Schema, List<SchemaPairCompatibility>> fullyCompatibleFn = (reader, writer) ->
+            List.of(backwardCompatibleFn.apply(reader, writer), forwardCompatibleFn.apply(reader, writer));
+
+
+    /**
+     * Evaluate schema evolution for a given list of Schemas and a new Schema provided a CompatibilityMode.
+     *
+     * @param original
+     * @param newSchema
+     * @param compatibilityMode
+     * @return List of Incompatibilities
+     */
+    public List<AvroIncompatibility> validateSchema(final List<Schema> original, final Schema newSchema,
+                                                    final CompatibilityMode compatibilityMode)
+                                                    throws AvroRuntimeException {
+        if (original == null || original.isEmpty()){
+            throw new IllegalArgumentException("previous schemas cannot be empty");
+        }
+        if (newSchema == null){
+            throw new IllegalArgumentException("new schema cannot be null");
+        }
+
+        final Supplier<Schema> getLastSchema = () -> original.get(original.size() - 1);
+
+        final Function<Stream<SchemaPairCompatibility>, List<AvroIncompatibility>>
+                flatMapToIncompat = inputStream -> inputStream.
+                        map(this::toAvroIncompatibility).
+                        flatMap(Collection::stream).collect(Collectors.toList());
+
+        switch (compatibilityMode) {
+            case FORWARD:
+                return forwardCompatibleFn.andThen(this::toAvroIncompatibility).
+                        apply(newSchema, getLastSchema.get());
+
+            case BACKWARD:
+                return backwardCompatibleFn.andThen(this::toAvroIncompatibility).
+                        apply(newSchema, getLastSchema.get());
+
+            case COMPATIBLE:
+                return flatMapToIncompat.apply(
+                        fullyCompatibleFn.apply(newSchema, getLastSchema.get()).stream()
+                );
+
+            case FORWARD_TRANSITIVE:
```

Comment:
I suggest to keep compatibility modes the same as it is for JSON to reduce confusion